data_creation/data_creation.py

```diff
@@ -0,0 +1,357 @@
+            
+            # Read the pose header from the dataset's predefined file
+            with open(dataset_module._POSE_HEADERS["holistic"], "rb") as buffer:
+                pose_header = PoseHeader.read(BufferReader(buffer.read()))
+            
+            # Cache the pose header
+            self._pose_headers_cache[dataset_name] = pose_header
+            return pose_header
+        except Exception as e:
+            print(f"Error getting pose header for {dataset_name}: {e}")
+            raise
+
+    def _create_pose_object(self, pose_datum, dataset_name: str):
+        """
+        Creates a Pose object from the given pose data_entry and dataset name.
+        :param pose_datum: Pose data_entry dictionary containing fps, data_entry, and conf.
+        :param dataset_name: Name of the dataset.
+        :return: Pose object.
+        """
+        try:
+            # Retrieve the pose header for the dataset
+            pose_header = self._get_pose_header(dataset_name)
+            
+            # Extract pose data_entry
+            fps = float(pose_datum["fps"].numpy())
+            pose_data = pose_datum["data"].numpy()
+            conf = pose_datum["conf"].numpy()
+            pose_body = NumPyPoseBody(fps, pose_data, conf)
+            
+            # Construct and return the Pose object
+            pose = Pose(pose_header, pose_body)
+            return pose
+        except Exception as e:
+            print(f"Error creating pose object for {dataset_name}: {e}")
+            raise
+
+    def _process_sentence(self, data_entry):
+        """
+        Processes a single sentence entry from the DGS Corpus.
+
+        :param data_entry: A single sentence entry from the DGS Corpus.
+        :return: Dictionary with sentence ID, updated pose, original pose, and metadata.
+        """
+        
+        # Sentence data
+        sentence = data_entry['sentence']
+        sentence_id = sentence['id'].numpy()
+        sentence_start_time = sentence['start'].numpy()
+        sentence_end_time = sentence['end'].numpy()
+
+        glosses = sentence['glosses']['gloss'].numpy()
+        gloss_start_times = sentence['glosses']['start'].numpy()
+        gloss_end_times = sentence['glosses']['end'].numpy()
+
+        # Pose data
+        pose_datum = data_entry['pose']
+        pose_data = pose_datum['data'].numpy() # (Frames, People, Points, Dims)
+        fps = pose_datum['fps'].numpy()
+        conf = pose_datum['conf'].numpy()
+        pose_body = NumPyPoseBody(fps, pose_data, conf)
+
+        original_poses_sequence = []
+        updated_poses_sequence = []
+
+        # Calculate sentence frame range
+        sentence_start_frame = math.floor(sentence_start_time / 1000 * fps)
+        sentence_end_frame = math.ceil(sentence_end_time / 1000 * fps)
+        sentence_frame_range = range(sentence_start_frame, sentence_end_frame)
+
+        # Construct metadata dictionary
+        metadata = {
```

Comment:
there's a lot of info here. maybe even too much.
all we **need** is the document id and entry id, sentence id (maybe sentence content, and time)
so that maybe if we want to know more, we can go to:
https://www.sign-lang.uni-hamburg.de/meinedgs/html/1413451-11105600-11163240_en.html 
and see the sentence in the original data.